src/utils.c

```diff
@@ -0,0 +1,46 @@
+#include <sys/time.h>
+#include <time.h>
+
+#include "utils.h"
+
+#if defined(__APPLE__)
+#define HAVE_MACH_TIMER
+#include <mach/mach_time.h>
+#elif !defined(_WIN32) && !defined(_WIN64)
+#define HAVE_POSIX_TIMER
+#ifdef CLOCK_MONOTONIC
+#define CLOCKID CLOCK_MONOTONIC
+#else
+#define CLOCKID CLOCK_REALTIME
+#endif
+#endif
+
+struct timeval rv_gettimeofday()
```

Comment:
You should specify a pointer to `struct timeval` as the necessary parameter in this function. Therefore, we can avoid unintended data copying.

Reply:
That is, the caller of `rv_gettimeofday` should prepare an object of `struct timeval`, and there should be no data on stack.